File: Graph/HashTable/3838-path-existence-queries-in-a-graph-i/path-existence-queries-in-a-graph-i.java
class Solution {
    public boolean[] pathExistenceQueries(int n, int[] nums, int maxDiff, int[][] queries) {
        int component[] = new int[n];
        int cmp =0;
        component[0] = cmp;
        for(int i=1;i<n;i++){
            if(nums[i] - nums[i-1] > maxDiff){
                cmp++;
            }
            component[i] = cmp;
        }
        int len = queries.length;
        boolean res[] = new boolean[len];
        for(int i=0;i<len;i++){
            int u = queries[i][0];
            int v = queries[i][1];
            res[i] = (component[u] == component[v]);
        }
        return res;
    }
}

File: Graph/HashTable/3838-path-existence-queries-in-a-graph-i/path-existence-queries-in-a-graph-i.py
class Solution:
    def pathExistenceQueries(self, n: int, nums: List[int], maxDiff: int, queries: List[List[int]]) -> List[bool]:
        component =[]
        cmpt = 0
        component.append(cmpt)
        for i in range(1,len(nums)):
            if nums[i] - nums[i-1] > maxDiff: cmpt +=1
            component.append(cmpt)
        res = []
        for u,v in queries:
            res.append(component[u] == component[v])
            
        return res

